integration-tests/multitenancy.test.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0 *
+ */
+
+import { AxiosInstance } from 'axios';
+import {
+    expectResourceToNotBePartOfSearchResults,
+    getFhirClient,
+    randomPatient,
+    waitForResourceToBeSearchable,
+} from './utils';
+import BulkExportTestHelper from './bulkExportTestHelper';
+
+jest.setTimeout(300_000);
+
+test('empty test placeholder', () => {
+    // empty test to avoid the "Your test suite must contain at least one test." error
+});
+
+if (process.env.MULTI_TENANCY_ENABLED === 'true') {
+    describe('tenant data isolation', () => {
```

Comment:
An interesting test case I would like to try out is:
tenant1: has Patient/1234
tenant2: has Observation/1234 with subject set to <url>/tenant1/Patient/1234

Then do searches to see if we can break the isolation (I am thinking includes & rev_includes)

It may be easy to use two obscure resourceTypes to avoid search results getting too big

Reply:
that's a good one. I added/revinclude tests with references to other tenant's resources.